From 9909f146b28d56c9c0411329a056ed959b33f76a Mon Sep 17 00:00:00 2001
From: Zequan Wu <zequanwu@google.com>
Date: Wed, 10 Feb 2021 03:26:00 +0000
Subject: [PATCH] Reland "Add [[clang::nomerge]] attribute to ~CheckError()."

This is a reland of 8860253376c38c090d585bda4b20b801e3aa3ce3

Original change's description:
> Add [[clang::nomerge]] attribute to ~CheckError().
>
> To disable merging multiple ~CheckError() destructor for accurate crash logs.
>
> Bug: 1153188
> Change-Id: If6d153661667a63f13b645f6d284eb3d5ea3a300
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2611331
> Commit-Queue: Nico Weber <thakis@chromium.org>
> Auto-Submit: Zequan Wu <zequanwu@google.com>
> Reviewed-by: Nico Weber <thakis@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#844989}

Bug: 1153188
Change-Id: I303c5ff9fb88f7a30663400622b327a910d0b108
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2686331
Reviewed-by: Nico Weber <thakis@chromium.org>
Commit-Queue: Nico Weber <thakis@chromium.org>
Commit-Queue: Zequan Wu <zequanwu@google.com>
Cr-Commit-Position: refs/heads/master@{#852453}
---
 base/check.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/check.h b/base/check.h
index c94ab68db90..17048e45539 100644
--- a/base/check.h
+++ b/base/check.h
@@ -85,7 +85,7 @@ class BASE_EXPORT CheckError {
   // Stream for adding optional details to the error message.
   std::ostream& stream();
 
-  ~CheckError();
+  NOMERGE ~CheckError();
 
   CheckError(const CheckError& other) = delete;
   CheckError& operator=(const CheckError& other) = delete;
